lattices/src/pair.rs

```diff
@@ -69,13 +71,52 @@ where
     }
 }
 
-impl<LatA, LatB> Default for Pair<LatA, LatB>
+impl<LatASelf, LatAOther, LatBSelf, LatBOther> PartialEq<Pair<LatAOther, LatBOther>>
+    for Pair<LatASelf, LatBSelf>
 where
-    LatA: Default,
-    LatB: Default,
+    LatASelf: PartialEq<LatAOther>,
+    LatBSelf: PartialEq<LatBOther>,
 {
-    fn default() -> Self {
-        let (a, b) = Default::default();
-        Self { a, b }
+    fn eq(&self, other: &Pair<LatAOther, LatBOther>) -> bool {
+        if self.a != other.a {
+            return false;
+        }
+
+        if self.b != other.b {
+            return false;
+        }
+
+        true
```

Comment:
Can just do `self.a == other.a && self.b == other.b` I think?

"a != b if and only if !(a == b)."

Reply:
yep, works.